### ISSUE TITLE
VBL and MBL colors are too similar

### ISSUE THREAD

**Issue by Azhrei** [RPTools/maptool#1623]
**Describe the bug**
VBL is purple when being drawn and MBL is blue(-er).  These are too close for easy distinguishing for some people.  (This is a high priority issue for my GM as VBL and MBL are almost indistinguishable for them.)

**Expected behavior**
Since red/green color blindness is the most common, the MBL should probably be magenta instead.

A more general solution would provide a default color palette in the Accessibility tab of Preferences and let the user pick the colors they want.  We could provide three or four default palettes for the most common forms of color blindness.

**Additional context**
This page has an interactive tool for demonstrating to people without color blindness what a colorblind individual might see:  https://davidmathlogic.com/colorblind/

**Comment by bubblobill** [comment]
Textured/dashed lines should also be considered. Easily achieved with Basic Stroke.

**Comment by JamzTheMan** [comment]
But MBL is magenta. Are you sure you are not confusing it when the two overlap? It does turn more purple then and is hard to see sometimes.

But I do think a config option for accessibly is warranted. Could that be done with themes? Can we add the three vbl/mbl options there?

FYI all of those colors (blue, magenta, and yellow) are defined in app constants so it's easy to change/override.

Another idea to help us to add buttons to turn on/off each layer and to be able to do this at any time not not during drawing

**Comment by Phergus** [comment]
I think he found a bug.  On a new campaign, with MBL selected, draw out various MBL shapes.  Note they all lay down in blue.  Test with token that has sight to see that they are indeed VBL not MBL.  Switch to VBL and then back to MBL and it works as expected.

**Comment by Phergus** [comment]
This probably explains why there were a couple mentions of tokens moving through MBL.

**Comment by JamzTheMan** [comment]
I can not replicate this in 1.6.1. 

1. Open MT, select VBLtools, tool is default to V, draws blue VBL.
2. Switch to M, draws in magenta MBL
3. Close MT, select VBL tools, tool default is now M (remembers last preference working)
4. Draws magenta MBL 

maybe I'm missing a step?

**Comment by Phergus** [comment]
Ah. Mine was defaulting to MBL.  I assume the last one I used?

**Comment by JamzTheMan** [comment]
Yea, on press of the button it's saved as an AppPreference so that persists.

Ah ha, ya, that's it, New Campaign resets and doesn't honor the button. (Probably a new issue should be opened for this). Also note multiple clients share AppPreferences so test accordingly (I did initially test this and should be ok but retest)

**Comment by Phergus** [comment]
Will open a ticket on the button issue.

**Comment by Phergus** [comment]
BTW, the topology colors are defined in AppStyle.java and are:

```Java
  public static Color topologyColor = new Color(0, 0, 255, 128);
  public static Color topologyTerrainColor = new Color(255, 0, 255, 128);
  public static Color tokenTopologyColor = new Color(255, 255, 0, 128);
```
Using the website above we see:
![image](https://user-images.githubusercontent.com/34379254/80754097-aea1ac00-8aeb-11ea-822d-dc7cef6d5bb2.png)

So VBL, MBL and TVBL should all be pretty distinguishable from each other.  It's when you have VBL and MBL overlapping that the combo is very close to VBL alone.

**Comment by Azhrei** [comment]
Good point, it's probably the overlapping that caught him.  I remember we were talking at the time about drawing VBL and MBL, so it definitely could've been the VBL vs. VBL+MBL that was difficult for him to tell apart.

Hm.  The additive aspects do make it more difficult to find colors that would work well for everyone.  Either adding them to a theme or using @bubblobill idea of textures instead seem like good approaches as it then leaves it up to the user what looks distinguishable to them (I mentioned red/green colorblindness because it's the most common, but there are numerous other combinations that are less common).

**Comment by Phergus** [comment]
If VBL used a pattern (say diagonal stripes up to the right) and MBL used the opposite pattern that could work for areas.  However both VBL and MBL are often drawn with single lines (or are the actually thin rectangles?) and so patterns might not work at all unless they are very fine.

**Comment by adventuremagic123** [comment]
Thank you for working on red/green color blindness issues.  I have that problem.

**Comment by JamzTheMan** [comment]
There really is no reason this could not be configurable and preference settings. Maybe we hook our draw panel selections and as such let you pick color or texture? Maybe we can add that panel in a way that it can be shown at any time and also has toggles to hide/show the different BL's?

**Comment by melek** [comment]
I admit I have the issue specifically that VBL and VBL+MBL are really hard to tell apart for me when overlaid on maps, especially when fog of war is displayed (and I'm not color blind!)

Is there a chance to simply pick a more contrasting combination (VBL+MBL) color in the short term until someone has the time to add the options to the UI? Maybe that orange (3rd down on Phergus' image list of colors) would be a good MBL+VBL color.
